### PR TITLE
Update to lsp-types 0.58.0

### PR DESCRIPTION
**Pull request by kjeremy**
Fixes CodeActionResponse

### CODE REVIEW

File: crates/ra_lsp_server/src/main_loop/handlers.rs [hunk 1]
```diff
@@ -666,7 +666,7 @@ pub fn handle_code_action(
 
     let assists = world.analysis().assists(FileRange { file_id, range })?.into_iter();
     let diagnostics = world.analysis().diagnostics(file_id)?;
-    let mut res: Vec<CodeAction> = Vec::new();
+    let mut res: CodeActionResponse = Default::default();
```
Comment (matklad) [inline crates/ra_lsp_server/src/main_loop/handlers.rs]:
```suggestion
    let mut res = CodeActionResponse::default();
```

File: crates/ra_lsp_server/src/main_loop/handlers.rs [hunk 2]
```diff
@@ -690,7 +690,7 @@ pub fn handle_code_action(
             edit: None,
             command: Some(command),
         };
-        res.push(action);
+        res.push(CodeActionOrCommand::CodeAction(action));
```
Comment (matklad) [inline crates/ra_lsp_server/src/main_loop/handlers.rs]: I wonder is there an `.into()` impl by any chance?

Comment (kjeremy) [reply crates/ra_lsp_server/src/main_loop/handlers.rs]: There does not appear to be.

Comment (kjeremy) [reply crates/ra_lsp_server/src/main_loop/handlers.rs]: I've submitted a PR for this to lsp-types.

**Pull request merged**